python/lsst/meas/base/applyApCorr.py

```diff
@@ -0,0 +1,159 @@
+    """
+    ConfigClass = ApplyApCorrConfig
+    _DefaultName = "applyApCorr"
+
+    def __init__(self, schema, **kwds):
+        lsst.pipe.base.Task.__init__(self, **kwds)
+
+        self.apCorrInfoDict = dict()
+        apCorrNameSet = getApCorrNameSet()
+        ignoreSet = set(self.config.ignoreList)
+        missingNameSet = ignoreSet - set(apCorrNameSet)
+        if missingNameSet:
+            self.log.warn("Fields in ignoreList that are not in fluxCorrectList: %s" %
+                (sorted(list(missingNameSet)),))
+        for algName in apCorrNameSet - ignoreSet:
+            if algName + "_flux" in schema:
+                self.apCorrInfoDict[algName] = ApCorrInfo(schema=schema, algName=algName)
+            else:
+                # if a field in the registry is missing from the schema, silently ignore it.
+                pass
+
+    def run(self, catalog, apCorrMap):
+        """Apply aperture corrections to a catalog of sources
+
+        @param[in,out] catalog  catalog of sources
+        @param[in] apCorrMap  aperture correction map (an lsst.afw.image.ApCorrMap)
+        """
+
+        for apCorrInfo in self.apCorrInfoDict.itervalues():
+            try:
+                apCorrModel = apCorrMap.get(apCorrInfo.fluxName)
+                apCorrSigmaModel = apCorrMap.get(apCorrInfo.fluxSigmaName)
+            except Exception:
+                self.log.warn("Could not find %s in apCorrMap" % (apCorrInfo.algName,))
+                for source in catalog:
+                    source.set(apCorrInfo.apCorrFlagKey, True)
+                continue
+
+            for source in catalog:
+                center = source.getCentroid()
+                # say we've failed when we start; we'll unset these flags when we succeed
+                source.set(apCorrInfo.apCorrFlagKey, True)
+                oldFluxFlagState = False
+                if self.config.doFlagApCorrFailures:
+                    oldFluxFlagState = source.get(apCorrInfo.fluxFlagKey)
+                    source.set(apCorrInfo.fluxFlagKey, True)
+
+                apCorr = 1.0
+                apCorrErr = 0.0
+                try:
+                    apCorr = apCorrModel.evaluate(center)
+                    apCorrErr = apCorrSigmaModel.evaluate(center)
+                except Exception:
+                    continue
+
+                source.set(apCorrInfo.apCorrKey, apCorr)
+                source.set(apCorrInfo.apCorrSigmaKey, apCorrErr)
+                if apCorr <= 0.0 or apCorrErr < 0.0:
+                    continue
+
+                flux = source.get(apCorrInfo.fluxKey)
+                fluxErr = source.get(apCorrInfo.fluxSigmaKey)
+                source.set(apCorrInfo.fluxKey, flux*apCorr)
+                a = flux/fluxErr
+                if apCorrErr == 0:
+                    b = float("inf")
+                else:
+                    b = apCorr/apCorrErr
+                source.set(apCorrInfo.fluxSigmaKey, abs(flux*apCorr)*math.sqrt(a*a + b*b))
```

Comment:
I realize this is what you inherited from the HSC side, but I think we want `apCorrErr==0` to lead to just `fluxSigma*=apCorr` (i.e. zero uncertainty in the aperture correction should mean we just scale the existing uncertainty by the correction factor).

By the way, this propagation of errors isn't actually correct - it includes some of the photon noise twice in the uncertainty, because it enters both through the aperture correction errors and the original flux errors.  But we don't actually have something better to change it to just yet (this is something that's still being figured out on the HSC side).  When we have that figured out, we'll probably change the formula entirely, but in the meantime we should get the zero-error case handled properly in case we decide it's actually better to set all the error to zero in an earlier stage.